cmd/flux/kustomization_test.go

```diff
@@ -0,0 +1,57 @@
+// +build e2e
+
+package main
+
+import "testing"
+
+func TestKustomizationFromGit(t *testing.T) {
+	cases := []struct {
+		args       string
+		goldenFile string
+	}{
+		{
+			"create source git tkfg --url=https://github.com/stefanprodan/podinfo --branch=main --tag=6.0.0",
+			"testdata/kustomization/create_source_git.golden",
+		},
+		{
+			"create kustomization tkfg --source=tkfg --path=./deploy/overlays/dev --prune=true --interval=5m --validation=client --health-check=Deployment/frontend.dev --health-check=Deployment/backend.dev --health-check-timeout=3m",
+			"testdata/kustomization/create_kustomization_from_git.golden",
+		},
+		{
+			"get kustomization tkfg",
+			"testdata/kustomization/get_kustomization_from_git.golden",
+		},
+		{
+			"reconcile kustomization tkfg --with-source",
+			"testdata/kustomization/reconcile_kustomization_from_git.golden",
+		},
+		{
+			"suspend kustomization tkfg",
+			"testdata/kustomization/suspend_kustomization_from_git.golden",
+		},
+		{
+			"resume kustomization tkfg",
+			"testdata/kustomization/resume_kustomization_from_git.golden",
+		},
+		{
+			"delete kustomization tkfg --silent",
+			"testdata/kustomization/delete_kustomization_from_git.golden",
+		},
+	}
+
+	namespace := "tkfg"
```

Comment:
I would use a func which adds a random suffix here (and in other places) to ensure accidental copy/paste without changes do not result in flakiness.

Reply:
We can't do this since the namespace is in the gold file.

Reply:
Light templating? But valid reason to not do this now.

Reply:
No idea how to do this templating, as the whitespace in tables changes based on length. I would unblock CI first then figure out if golden files that are templates make sense.